src/plugins/intel_cpu/src/nodes/mkldnn_non_zero.cpp

```diff
@@ -108,32 +115,129 @@ void MKLDNNNonZeroNode::executeSpecified() {
+                               dst[outputIndex] = static_cast<int>(i0);
+                               dst[outputIndex + totalNonZeroCount] = static_cast<int>(i1);
+                               dst[outputIndex + x2totalNonZeroCount] = static_cast<int>(i2);
+                               outputIndex++;
+                           }
+                       });
+        break;
+    }
+    case 4:
+    {
+        size_t i2Stride = srcDims[3];
+        size_t i1Stride = srcDims[2] * i2Stride;
+        size_t i0Stride = srcDims[1] * i1Stride;
+        size_t x2totalNonZeroCount = totalNonZeroCount * 2;
+        size_t x3totalNonZeroCount = totalNonZeroCount * 3;
+
+        parallel_for4d(srcDims[0], srcDims[1], srcDims[2], srcDims[3],
+                       [&](size_t ithr, size_t i0, size_t i1, size_t i2, size_t i3) {
+                           size_t inputIndex = i0 * i0Stride + i1 * i1Stride + i2 * i2Stride + i3;
+                           size_t& outputIndex = destIndices[ithr];
+
+                           if (src[inputIndex] != zero) {
+                               dst[outputIndex] = static_cast<int>(i0);
+                               dst[outputIndex + totalNonZeroCount] = static_cast<int>(i1);
+                               dst[outputIndex + x2totalNonZeroCount] = static_cast<int>(i2);
+                               dst[outputIndex + x3totalNonZeroCount] = static_cast<int>(i3);
+                               outputIndex++;
+                           }
+                       });
+        break;
+    }
+    case 5:
+    {
+        size_t i3Stride = srcDims[4];
+        size_t i2Stride = srcDims[3] * i3Stride;
+        size_t i1Stride = srcDims[2] * i2Stride;
+        size_t i0Stride = srcDims[1] * i1Stride;
+        size_t x2totalNonZeroCount = totalNonZeroCount * 2;
+        size_t x3totalNonZeroCount = totalNonZeroCount * 3;
+        size_t x4totalNonZeroCount = totalNonZeroCount * 4;
+
+        parallel_for5d(srcDims[0], srcDims[1], srcDims[2], srcDims[3], srcDims[4],
+                       [&](size_t ithr, size_t i0, size_t i1, size_t i2, size_t i3, size_t i4) {
+                           size_t inputIndex = i0 * i0Stride + i1 * i1Stride + i2 * i2Stride + i3 * i3Stride + i4;
+                           size_t& outputIndex = destIndices[ithr];
+
+                           if (src[inputIndex] != zero) {
+                               dst[outputIndex] = static_cast<int>(i0);
+                               dst[outputIndex + totalNonZeroCount] = static_cast<int>(i1);
+                               dst[outputIndex + x2totalNonZeroCount] = static_cast<int>(i2);
+                               dst[outputIndex + x3totalNonZeroCount] = static_cast<int>(i3);
+                               dst[outputIndex + x4totalNonZeroCount] = static_cast<int>(i4);
+                               outputIndex++;
+                           }
+                       });
+        break;
+    }
+    default:
+        assert(false);
```

Comment:
There is no such restriction in the specification. Add at least old approach with parallel_for.

Reply:
Done